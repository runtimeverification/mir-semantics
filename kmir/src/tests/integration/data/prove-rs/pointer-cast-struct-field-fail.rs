struct Pair {
    head: i64,
    tail: i8,
}

fn read_via_cast(pair: &Pair) -> i64 {
    unsafe { *(pair as *const Pair as *const i64) }
}

fn main() {
    let pair = Pair { head: 7, tail: 1 };
    assert!(read_via_cast(&pair) == 7);
}
