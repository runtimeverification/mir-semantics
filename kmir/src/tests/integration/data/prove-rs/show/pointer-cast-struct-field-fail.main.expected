
┌─ 1 (root, init)
│   #execTerminator ( terminator ( ... kind: terminatorKindCall ( ... func: operandC
│   span: src/rust/library/std/src/rt.rs:194
│
│  (135 steps)
├─ 3
│   #expect ( thunk ( #applyBinOp ( binOpEq , thunk ( #applyBinOp ( binOpBitAnd , th
│   function: read_via_cast
┃
┃ (1 step)
┣━━┓
┃  │
┃  ├─ 4
┃  │   AssertError ( assertMessageMisalignedPointerDereference ( ... required: operandC
┃  │   function: read_via_cast
┃  │
┃  │  (1 step)
┃  └─ 6 (stuck, leaf)
┃      #ProgramError ( AssertError ( assertMessageMisalignedPointerDereference ( ... re
┃      function: read_via_cast
┃
┗━━┓
   │
   ├─ 5
   │   #execBlockIdx ( basicBlockIdx ( 1 ) ) ~> .K
   │   function: read_via_cast
   │
   │  (7 steps)
   └─ 7 (stuck, leaf)
       #traverseProjection ( toLocal ( 2 ) , thunk ( #cast ( PtrLocal ( 1 , place ( ...
       function: read_via_cast
       span: r/prove-rs/pointer-cast-struct-field-fail.rs:8


┌─ 2 (root, leaf, target, terminal)
│   #EndProgram ~> .K



